File: packages/sdui-common/src/main/java/nl/q42/common/components/TextComponent.java
package nl.q42.common.components;

public class TextComponent extends Component
{

  public TextColor      color;
  public TextFormatting formatting;
  public String         text;

  public TextComponent(
      String text,
      TextColor color,
      TextFormatting formatting,

      String contentId)
  {
    super(ComponentTypes.TEXT, contentId);
    this.formatting = formatting;
    this.color      = color;
    this.text       = text;
  }

  public TextComponent(String text, String contentId)
  {
    this(text, TextColor.PRIMARY, TextFormatting.NORMAL, contentId);
  }
}

File: packages/sdui-common/src/main/kotlin/nl/q42/common/components/TextComponent.kt
package nl.q42.common.components

class TextComponent(
    var text: String?,
    var color: TextColor?,
    var formatting: TextFormatting?,

    contentId: String?
) : Component(ComponentTypes.TEXT, contentId) {
    constructor(text: String?, contentId: String?) : this(
        text,
        TextColor.PRIMARY,
        TextFormatting.NORMAL,
        contentId
    )
}
